docs/source/challenge_creation.md

```diff
@@ -0,0 +1,195 @@
+
+  A dataset in EvalAI is the main entity on which an AI challenge is based on. Participants are expected to make submissions corresponding to different splits of the corresponding dataset. A dataset is generally divided into different parts and each part is called dataset split. Generally, a dataset has three different splits:
+
+  1. Training set
+  2. Validation set
+  3. Test set
+
+  * **id**: Unique integer identifier for the dataset split
+
+  * **name**: Name of the dataset split (it must be unique for every dataset split)
+
+  * **codename**: Codename of dataset split. Note that the codename of dataset split is used to map the results returned by the evaluation script to a particular dataset split in EvalAI's database. Please make sure that no two dataset splits have the same codename. Again, make sure that the dataset split's codename match with what is there in the evaluation script provided by the challenge host.
+
+  
+* **challenge_phase_splits**:
+
+  A challenge phase split is a relation between a challenge phase and dataset splits for a challenge (many to many relation). This is used to set the privacy of submissions (public/private) to different dataset splits for different challenge phases.
+
+  * **challenge_phase_id**: Id of challenge_phase (Gets the challenge phase details to map with)
+
+  * **leaderboard_id**: Id of leaderboard (Given above)
+
+  * **dataset_split_id**: Id of dataset split (Given above)
+
+  * **visibility**: Enter any of the positive integers given below.
+
+    a) HOST -> 1
+
+    b) OWNER AND HOST -> 2
+
+    c) PUBLIC -> 3
+
+### Sample zip configuration file
+
+Here is the sample configuration file for a challenge with 1 phase and 2 dataset split:
+
+```yaml
+
+title: Challenge Title
+short_description: Short description of the challenge (preferably 140 characters)
+description: description.html
+evaluation_details: evaluation_details.html
+terms_and_conditions: terms_and_conditions.html
+image : logo.jpg
+submission_guidelines: submission_guidelines.html
+evaluation_script: evaluation_script.zip
+start_date: 2017-06-09 20:00:00
+end_date: 2017-06-19 20:00:00
+published: True
+
+leaderboard:
+  - id: 1
+    schema: {"labels": ["yes/no", "number", "others", "overall"], "default_order_by": "overall"}
+  - id: 2
+    schema: {"labels": ["yes/no", "number", "others", "overall"], "default_order_by": "yes/no"}
+
+challenge_phases:
+  - id: 1
+    name: Challenge Name of the challenge phase
+    description: challenge_phase_description.html
+    leaderboard_public: True
+    is_public: True
+    start_date: 2017-06-09 20:00:00
+    end_date: 2017-06-19 20:00:00
+    test_annotation_file: test_annotation.txt
+    codename: Challenge phase codename
+    max_submissions_per_day: 100
+    max_submissions: 1000
+
+dataset_splits:
+  - id: 1
+    name: Name of the dataset split
+    codename: codename of dataset split
+  - id: 2
+    name: Name of the dataset split
+    codename: codename of dataset split
```

Comment:
No, we don't want to give the capability to edit the dataset split codename later.